--- git.js	2016-12-20 20:01:10.000000000 +1300
+++ git_cygwin.js	2017-01-15 04:18:50.710192700 +1300
@@ -11,13 +11,23 @@
 var git = npm.config.get('git')
 var assert = require('assert')
 var log = require('npmlog')
+var win32 = process.platform === "win32"
+var cygwin = win32 && (process.env.ORIGINAL_PATH || '').indexOf('/cygdrive/') != -1
 var noProgressTillDone = require('./no-progress-while-running.js').tillDone
 
 function prefixGitArgs () {
-  return process.platform === 'win32' ? ['-c', 'core.longpaths=true'] : []
+  // return process.platform === 'win32' ? ['-c', 'core.longpaths=true'] : []
+  return win32 ? ["-c", "core.longpaths=true"] : []
 }
 
 function execGit (args, options, cb) {
+  if(cygwin && args) {
+    for(var i=0; i<args.length; i++) {
+      if(':\\'.indexOf(args[i]) != 1) {
+        args[i] = args[i].replace(/\\/g, '/').replace(/^([A-Za-z])\:\//, '/cygdrive/$1/');
+      }
+    }
+  }
   log.info('git', args)
   var fullArgs = prefixGitArgs().concat(args || [])
   return exec(git, fullArgs, options, noProgressTillDone(cb))
